### PR TITLE
model: support cosmos-predict2/predict2.5/transfer2.5 pipeline

### PR DESCRIPTION
**Pull request by rebel-seinpark**
# Pull Request Description

> **⚠️ Important: Branch Target**
> - **New features, enhancements, and non-critical fixes**: Merge to `dev` branch
> - **Critical hotfixes only**: Merge to `main` branch (must also merge to `dev`)
> 
> Please ensure you've selected the correct base branch before submitting!

## Type of Change

- [ ] Release (dev → main merge for production release)
- [ ] New Model Support
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Other (please describe):

## Related Issues

----
# Conventional commit
```
type(optional scope): description
```
----
# Type candidate
  - Model Updates
    - `model`: Adding New models or Bugfix for existing models
      - ex) Add LlavaNext 
      - ex) Bugfix Whisper
  - Enhancements
    - `performance`: Optimizing some models or this library itself
      - ex) Loading RBLNModel faster
      - ex) Optimizing Memory Usage of DecoderOnlyModel
  - Code Refactor
    - `refactor`: Re-arrange class architecture, or more.
      - ex) Refactor Seq2Seq
  - Documentation
    - `doc`: Update docstring only
  - Library Dependencies
    - `dependency`: Update requirements, something like that.
  - Release
    - `release`: Merging dev to main for production release
      - ex) Release v1.2.0
  - Other
    - `other`: None of above.
      - ex) ci update
      - ex) pdm update

### CODE REVIEW

File: src/optimum/rbln/diffusers/modeling_diffusers.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def from_pretrained(
                         raise ValueError(
                             f"Invalid module_name '{module_name}' found in model_index.json for "
                             f"submodule '{submodule_name}'. "
-                            "Expected 'optimum.rbln'. Please check the model_index.json configuration."
+                            "Expected 'optimum.rbln'. Please check the model_index.json configuration. "
```
Comment (rebel-seinpark) [inline src/optimum/rbln/diffusers/modeling_diffusers.py]:
```suggestion
                            "Expected 'optimum.rbln'. Please check the model_index.json configuration."
```

File: src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+        self.uses_encoder = uses_encoder
+        # No size defaults here: each pipeline config supplies its own diffusers-default
+        # height/width/num_frames, and a standalone compile must pass them explicitly.
+        self.num_frames = num_frames
+        self.height = height
+        self.width = width
+
+        # The pipeline hook sets this to the transformer's out_channels; a standalone compile
+        # derives it from the VAE z_dim in _update_rbln_config.
```
Comment (rebel-seinpark) [inline src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py]:
```suggestion
```

File: src/optimum/rbln/diffusers/configurations/models/configuration_transformer_cosmos.py [hunk 3]
```diff
@@ -46,12 +46,16 @@ def __init__(
...
             max_seq_len (int | None): Maximum sequence length of prompt embeds.
             embedding_dim (int | None): Embedding vector dimension of prompt embeds.
             num_channels_latents (int | None): The number of channels in latent space.
             latent_height (int | None): The height in pixels in latent space.
             latent_width (int | None): The width in pixels in latent space.
+            uses_per_frame_timestep (bool | None): Whether the pipeline feeds a per-frame timestep
+                tensor ([B, 1, T, 1, 1]; conditioning frames at t=0) instead of a per-batch scalar.
+                Decides the compiled shapes of embedded_timestep/temb. When None, inferred from the
+                model config (Cosmos-Predict2.5, fingerprinted by use_crossattn_projection, is
+                always per-frame; older families default to per-batch).
```
Comment (rebel-seinpark) [inline src/optimum/rbln/diffusers/configurations/models/configuration_transformer_cosmos.py]:
```suggestion
```

File: src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        self,
+        batch_size: int | None = None,
+        uses_encoder: bool | None = None,
+        num_frames: int | None = None,
+        height: int | None = None,
+        width: int | None = None,
+        num_channels_latents: int | None = None,
+        vae_scale_factor_temporal: int | None = None,
+        vae_scale_factor_spatial: int | None = None,
+        use_slicing: bool | None = None,
```
Comment (rebel-dkhong) [inline src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py]: In `RBLNAutoencoderKLWanConfig`, users can set `use_slicing`, but it never takes effect.
Also, `use_slicing` is always set to be `False` in `__post_init__` of `RBLNAutoencoderKLWan`.
If it should be `False`, how about removing the variable from this config class?

File: src/optimum/rbln/diffusers/models/autoencoders/autoencoder_kl_wan.py [hunk 5]
```diff
@@ -0,0 +1,770 @@
...
+            z = self.post_quant_conv.to(z.dtype)(z)
+        _, _, num_frame, _, _ = z.shape
+        war_kw = {"war_zero": torch.zeros(1, dtype=self.rbln_config.dtype)} if _EN_WAR else {}
+        outs = []
+        feat_cache_0 = None
+        for i in range(num_frame):
+            if i == 0:
+                ret = self.decoder_0(z[:, :, :1, :, :])
+            else:
+                ret = self.decoder_n(z[:, :, i : i + 1, :, :], feat_cache_0, **war_kw)
```
Comment (rebel-dkhong) [inline src/optimum/rbln/diffusers/models/autoencoders/autoencoder_kl_wan.py]: `decoder_0` and `decoder_n` are instances of `RBLNRuntimeWanVAEDecoder`, and the class has `use_slicing` logic in the `decode` method.
However, `decoder_0` and `decoder_n` are called directly, instead of using `decode` method.
So, I'm afraid the current implementation doesn't support batch slicing.

If you want to remove batch slicing logic, please see my other comment about `use_slicing` in the config class. If not, please update these call sites.

There is a similar problem in the encoder, so please take a look at it if you decide to support batch slicing.

File: src/optimum/rbln/diffusers/models/autoencoders/vae.py [hunk 6]
```diff
@@ -57,6 +57,31 @@ def decode(self, z: torch.FloatTensor, **kwargs) -> torch.FloatTensor:
...
+    """Runtime wrapper for Wan VAE encoder inference."""
+
+    def encode(self, x: torch.FloatTensor, **kwargs) -> torch.FloatTensor:
+        if self.use_slicing and x.shape[0] > 1:
+            encoded_slices = [self.forward(x_slice) for x_slice in x.split(1)]
+            h = torch.cat(encoded_slices)
+        else:
+            h = self.forward(x)
+        posterior = DiagonalGaussianDistribution(h)
+        return posterior
```
Comment (rebel-dkhong) [inline src/optimum/rbln/diffusers/models/autoencoders/vae.py]: The `encode` method in this class is not used - `RBLNAutoencoderKLWan` calls `encoder_0` / `encoder_n` directly, so they go through `forward`.
In particular, `DiagonalGaussianDistribution` is applied after the whole video (or image) is encoded.
Please see L695 in `autoencoder_kl_wan.py` and check whether one of them should be removed.

File: src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py (hunk 4 above)
Comment (rebel-seinpark) [reply src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py]: You are right. I fixed it. Thank you

File: src/optimum/rbln/diffusers/models/autoencoders/vae.py (hunk 6 above)
Comment (rebel-seinpark) [reply src/optimum/rbln/diffusers/models/autoencoders/vae.py]: Rather than deleting the dead encode/decode, the chunk split now lives where it belongs: `RBLNRuntimeWanVAEEncoder` wraps the E0+EN graphs and `RBLNRuntimeWanVAEDecoder` wraps D0+DN (plus the host-side `post_quant_conv`), each owning its chunk loop - so encode/decode are the real implementations and the model exposes the upstream encoder/decoder shape with _encode/_decode as one-line delegates.

File: src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py (hunk 4 above)
Comment (rebel-seinpark) [reply src/optimum/rbln/diffusers/configurations/models/configuration_autoencoder_kl_wan.py]: I first removed the field as suggested (b75b0501), but digging further showed slicing
is not an optional memory knob here — the Wan VAE graphs are *always* compiled at
batch_size=1 (the decoder's working set is near the device limit), so per-sample
execution is the only way to serve a batched request. The field is therefore back with
real semantics, settled in 0e53197a:

- `use_slicing` defaults to `True` and is honored end to end: the model attribute
  reflects it truthfully, and both `encode` and `decode` slice batched inputs
  (`decode` was missing its slicing branch entirely — restored).
- The stored `batch_size` is always normalized to 1 with a warning, so the saved
  `rbln_config.json` describes the compiled artifact rather than the request; a
  batch>1 request with `use_slicing=False` additionally flips slicing on.
- `use_slicing=False` at batch 1 remains a valid opt-out.

File: src/optimum/rbln/diffusers/models/autoencoders/autoencoder_kl_cosmos.py [hunk 7]
```diff
@@ -127,6 +126,8 @@ def replaced_forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
     def update_rbln_config_using_pipe(
         cls, pipe: "RBLNDiffusionMixin", rbln_config: "RBLNDiffusionMixinConfig", submodule_name: str
     ) -> "RBLNDiffusionMixinConfig":
+        # out_channels is the pure latent channel count for every Cosmos family; in_channels may
+        # carry an extra condition-mask channel depending on the pipeline.
```
Comment (rebel-seinpark) [inline src/optimum/rbln/diffusers/models/autoencoders/autoencoder_kl_cosmos.py]:
```suggestion
```

File: src/optimum/rbln/diffusers/models/controlnets/controlnet_cosmos.py [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+        # 5. Timestep embeddings (Transfer2.5 feeds per-frame timesteps, [B, 1, T, 1, 1]).
+        if timestep.ndim == 1:
+            temb, embedded_timestep = self.time_embed(base_hidden_states, timestep)
+        elif timestep.ndim == 5:
+            assert timestep.shape == (B, 1, T, 1, 1), (
+                f"Expected timestep to have shape [B, 1, T, 1, 1], but got {timestep.shape}"
+            )
+            temb, embedded_timestep = self.time_embed(base_hidden_states, timestep.flatten())
+            # Keep the embeddings per-frame: the graph broadcasts them over the spatial tokens
+            # (see CosmosPerFrameAdaLNMixin), so the H*W-fold expansion never crosses into it.
```
Comment (rebel-seinpark) [inline src/optimum/rbln/diffusers/models/controlnets/controlnet_cosmos.py]:
```suggestion
            # Keep the embeddings per-frame
```

File: src/optimum/rbln/diffusers/models/controlnets/controlnet_cosmos.py [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+            # here too; the graph broadcasts them over the spatial tokens.
+            ("embedded_timestep", [rbln_config.batch_size, num_frames, hidden_size], rbln_config.dtype),
+            ("temb", [rbln_config.batch_size, num_frames, hidden_size * 3], rbln_config.dtype),
+            ("image_rotary_emb_0", [hidden_dim, model_config.attention_head_dim], "float32"),
+            ("image_rotary_emb_1", [hidden_dim, model_config.attention_head_dim], "float32"),
+            ("conditioning_scales", [n_blocks], rbln_config.dtype),
+        ]
+        if model_config.img_context_dim_in is not None and model_config.img_context_dim_in > 0:
+            if rbln_config.img_context_num_tokens is None:
+                # The token count lives on the transformer config (the pipeline hook copies it);
```
Comment (rebel-seinpark) [inline src/optimum/rbln/diffusers/models/controlnets/controlnet_cosmos.py]:
```suggestion
```

File: src/optimum/rbln/diffusers/models/controlnets/controlnet_cosmos.py [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+        input_info = [
+            ("control_hidden_states", [rbln_config.batch_size, hidden_dim, hidden_size], rbln_config.dtype),
+            ("base_hidden_states", [rbln_config.batch_size, hidden_dim, hidden_size], rbln_config.dtype),
+            (
+                "encoder_hidden_states",
+                [rbln_config.batch_size, rbln_config.max_seq_len, rbln_config.embedding_dim],
+                rbln_config.dtype,
+            ),
+            # Transfer2.5 always feeds per-frame timesteps ([B, 1, T, 1, 1]). They stay per-frame
+            # here too; the graph broadcasts them over the spatial tokens.
```
Comment (rebel-seinpark) [inline src/optimum/rbln/diffusers/models/controlnets/controlnet_cosmos.py]:
```suggestion
            # Transfer2.5 always feeds per-frame timesteps ([B, 1, T, 1, 1]).
```
